File: stringDifference.go
package main

import ("fmt"
        "strings"
        "strconv")

func findTheDifference(s string, t string) byte {
  stringS := strings.Split(s, "")
  stringT := strings.Split(t, "")
  sMap := make(map[string]int)
  tMap := make(map[string]int)

  for _,li := range stringS {
      sMap[li] = sMap[li] + 1
    }
  for _,li := range stringT {
      tMap[li] = tMap[li] + 1
    }
    fmt.Println(sMap)
    fmt.Println(tMap)

  for _,li := range stringT {
      if sMap[li] != tMap[li]{
        answer := []byte(li)
        return answer
      }
    }
    return answer
}

func main() {
  s := "abcd"
  t := "abcde"

  fmt.Println(findTheDifference(s, t))
}

File: stringDifference.py
class Solution(object):
    def findTheDifference(self, s, t):

        stringS = list(s)
        print(stringS)
        stringT = list(t)
        print(stringT)
        answer = [x for x in stringT if x not in stringS]
        if len(answer) == 0:
            return s
        return answer[0]

sol = Solution()
s = "ae"
t = "aea"

ans = sol.findTheDifference(s,t)
print(ans)
